contracts/Super721.sol

```diff
@@ -1097,10 +1097,6 @@ contract Super721 is PermitControl, ERC165Storage, IERC721 {
   }
 
   function tokenURI(uint256 _tokenId) public view returns (string memory) {
-    return Strings.strConcat(
-        metadataUri,
-        Strings.uint2str(_tokenId),
-        ".json"
-    );
+    return metadata[_tokenId];
```

Comment:
Ah, sorry if the prompt was confusing: this is not what we wanted. It looks like what you've got here is returning the (completely optional) on-chain metadata that corresponds to a particular `_tokenId` instead of the bogus tokenURI.

We still do want to return a token URI instead of straight token metadata, ERC-721 just dictates that we need to return a fully-fledged URI with no client substitution required at all. Basically, we want to be able to specify `metadataUri` as a string containing some `{id}` substring, and then have that substring get filled in with a 64-character zero-padded hexadecimal string of `_tokenId` wherever that `{id}` tag may appear.

Make sense?